pandas/io/parsers.py

```diff
@@ -2251,6 +2314,113 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
         return values
 
 
+class BytesIOWrapper:
+    """
+    Allows the pyarrow engine for read_csv() to read from string buffers
+    """
+
+    def __init__(self, string_buffer: StringIO, encoding: str = "utf-8"):
+        self.string_buffer = string_buffer
+        self.encoding = encoding
+
+    def __getattr__(self, attr: str):
+        return getattr(self.string_buffer, attr)
+
+    def read(self, size: int = -1):
+        content = self.string_buffer.read(size)
+        return content.encode(self.encoding)
+
+
+class ArrowParserWrapper(ParserBase):
+    """
+    Wrapper for the pyarrow engine for read_csv()
+    """
+
+    def __init__(self, src, **kwds):
+        self.kwds = kwds
+        self.src = src
+
+        ParserBase.__init__(self, kwds)
+
+        encoding = kwds.get("encoding") if kwds.get("encoding") is not None else "utf-8"
+
+        self.usecols, self.usecols_dtype = _validate_usecols_arg(kwds["usecols"])
+        na_values = kwds["na_values"]
+        if isinstance(na_values, dict):
+            raise ValueError(
+                "The pyarrow engine doesn't support passing a dict for na_values"
+            )
+        self.na_values = list(
+            _clean_na_values(
+                kwds["na_values"], keep_default_na=kwds["keep_default_na"]
+            )[0]
+        )
+        if isinstance(self.src, TextIOBase):
+            self.src = BytesIOWrapper(self.src, encoding=encoding)
+
+    def read(self):
+        pyarrow = import_optional_dependency(
+            "pyarrow.csv",
+            min_version="0.15.0",
+            extra="pyarrow is required to use the pyarrow engine",
+        )
+        kwdscopy = {k: v for k, v in self.kwds.items() if v is not None}
+        # these are kwargs passed to pyarrow
+        parseoptions = {"delimiter", "quote_char", "escape_char", "ignore_empty_lines"}
+        convertoptions = {
+            "include_columns",
+            "null_values",
+            "true_values",
+            "false_values",
+        }
+        # rename some arguments to pass to pyarrow
+        kwdscopy["include_columns"] = kwdscopy.get("usecols")
+        kwdscopy["null_values"] = kwdscopy.get("na_values")
+        kwdscopy["escape_char"] = kwdscopy.get("escapechar")
+        kwdscopy["ignore_empty_lines"] = kwdscopy.get("skip_blank_lines")
+
+        parse_options = {k: v for k, v in kwdscopy.items() if k in parseoptions}
+        convert_options = {k: v for k, v in kwdscopy.items() if k in convertoptions}
+        headerexists = True if self.header is not None else False
```

Comment:
the whole structure of this class is really odd. The class should do consruction, option validation, reading, and post-processing by calling methods.